### PR TITLE
Remote

### PR DESCRIPTION
**Pull request by shubh90**

### CODE REVIEW

File: deploy/deploy.go [hunk 1]
```diff
@@ -451,4 +481,5 @@ func (m *MeshnetSpec) Healthy(ctx context.Context) error {
 			}
 		}
 	}
+
```
[the remark this answers is not in the document]
Comment (shubh90) [reply deploy/deploy.go]: Done

File: deploy/deploy.go [hunk 2]
```diff
@@ -39,74 +26,121 @@ import (
...
 const (
 	dockerConfigEnvVar           = "DOCKER_CONFIG"
 	kubeletConfigPathTemplate    = "%s:/var/lib/kubelet/config.json"
 	dockerConfigTemplateContents = `{
-  "auths": {
-{{range $val := .}}    "{{$val}}": {}
-{{end}}  }
-}
+		"auths": {
+	  {{range $val := .}}    "{{$val}}": {}
```
[the remark this answers is not in the document]
Comment (shubh90) [reply deploy/deploy.go]: Fixed

File: deploy/deploy.go [hunk 3]
```diff
@@ -415,9 +447,7 @@ func (m *MeshnetSpec) SetKClient(c kubernetes.Interface) {
 }
 
 func (m *MeshnetSpec) Deploy(ctx context.Context) error {
-	mPath := filepath.Join(deploymentBasePath, m.ManifestDir)
-	log.Infof("Deploying Meshnet from: %s", mPath)
```
[the remark this answers is not in the document]
Comment (shubh90) [reply deploy/deploy.go:419]: Done

File: deploy/deploy_test.go [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/fake"
+	fakecorev1 "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	ktest "k8s.io/client-go/testing"
+)
+
```
[the remark this answers is not in the document]
Comment (shubh90) [reply deploy/deploy_test.go:28]: So all of the methods in it like KindSpec, Metallb, Meshnet etc are unit tested already. I will get back to add this one more test after I finish the bigger implementations.

**Review by alexmasi — APPROVED** [review]

**Pull request merged**